cpp/include/proton/tag.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef PROTON_TAG_HPP
+#define PROTON_TAG_HPP
+
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "./fwd.hpp"
+#include "./internal/export.hpp"
+#include "./endpoint.hpp"
+#include "./receiver.hpp"
+#include "./sender.hpp"
+
+#include <string>
+
+/// @file
+/// @copybrief proton::tag
+
+// struct pn_delivery_tag_t;
+
+namespace proton {
+
+/// A container of tag.
+class
+PN_CPP_CLASS_EXTERN tag : public internal::object<pn_delivery_tag_t> {
+  public:
+    /// @cond INTERNAL
+    PN_CPP_EXTERN tag(pn_delivery_tag_t* t) : internal::object<pn_delivery_tag_t>(t) {}
```

Comment:
You should be able to include the header that defines the type, I added the following on top of the file and the compile error went away

```
#include "proton/delivery.h"
```

You can include C headers in a C++ code like this, because the delivery.h header has the following code on top, https://docs.microsoft.com/en-us/cpp/cpp/extern-cpp?view=msvc-160#extern-c-and-extern-c-function-declarations
```
#ifdef __cplusplus
extern "C" {
#endif
```